core/store/src/archive/cloud_storage/download.rs

```diff
@@ -59,3 +67,12 @@ impl CloudStorage {
         Ok(files.contains(&name))
     }
 }
+
+/// Deserializes raw bytes into a typed value using Borsh format.
+fn deserialize<T: BorshDeserialize>(
```

Comment:
Would derive deserialize on `CloudStorageFileID` be enough instead of writing this function?